src/prover_disk.hpp

```diff
@@ -241,6 +242,26 @@ class DiskProver {
     std::vector<uint64_t> table_begin_pointers;
     std::vector<uint64_t> C2;
 
+    // Using this method instead of simply seeking will prevent segfaults that would arise when
+    // continuing the process of looking up qualities.
+    static void SafeSeek(std::ifstream& disk_file, uint64_t seek_location) {
+        disk_file.seekg(seek_location);
+
+        if (disk_file.bad() || disk_file.fail()) {
```

Comment:
This is equivalent to checking only `.fail()`, as it returns true when either of `failbit` or `badbit` is set.

Reply:
done